nodejs/week3/warmup/index.js

```diff
@@ -0,0 +1,54 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+// Parse URL-encoded bodies (as sent by HTML forms)app.use(express.urlencoded({ extended: true }));// Parse JSON bodies (as sent by API clients)app.use(express.json());
+app.use("/calculator", router);
+router.get("/:operator", async (request, response) => { 
+    let operator = request.params.operator;
+ function calculator(operation) { 
+     firstParam = request.query.firstParam;
+      
+ let totalOfFirstParameter; 
+ if (typeof firstParam === "string" && firstParam !== "") { 
+     totalOfFirstParameter = parseInt(firstParam);
+ } else if (typeof firstParam === "object" && firstParam !== "" && !firstParam.includes("")) { 
+     totalOfFirstParameter = firstParam.map(Number). reduce((accumulator, firstParam) => operation(accumulator, firstParam));
```

Comment:
I think what you wanted to do in this case, is something like:
```
Object.values(req.query)
```
This is a nice way of reading all of the query parameters(first,second,third...) regardless of their name also.
Now you are reading the 'firstParam', and then trying to parse it as a list of parameters I guess, but that won't work!
So, what I suggest is you refactor it a bit to be more like this:
```
app.get("/calculator/add", (req, res) => {
    const queryParameters = Object.values(req.query).map(num => Number(num))
    const addition = queryParameters.reduce((previousValue, currentValue) => previousValue + currentValue)
})
```
However!! If you have tested that more than two parameters work you don't have to change it now, just keep it mind for the future :)